app.js

```diff
@@ -0,0 +1,468 @@
+
+      const filesInDelta = delta
+        .getInsertsFor(
+          triple(
+            undefined,
+            new NamedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
+            new NamedNode(
+              'http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#FileDataObject',
+            ),
+          ),
+        )
+        .map((insert) => insert.subject.value);
+
+      const logicalFilesInDelta = filesInDelta.filter(
+        (fileIRI) => fileIRI.slice(0, 8) !== 'share://',
+      );
+
+      if (!logicalFilesInDelta.length) {
+        console.log(
+          'No FileDataObject inserts for logical files. Nothing should happen.',
+        );
+        return res.status(204).send();
+      }
+
+      res.status(202).send();
+
+      const filesToScan = [...new Set(logicalFilesInDelta)]; //make them unique
+
+      console.log('File IRIs to be scanned: ' + JSON.stringify(filesToScan));
+
+      const fileResults = [];
+
+      for (const file of filesToScan) {
+        const scanFileResult = await scanFile(file);
+
+        const storeResult = await storeMalwareAnalysis(
+          file,
+          scanFileResult.stixMalwareAnalysis,
+        );
+
+        fileResults.push({
+          file,
+          ...scanFileResult,
+          ...storeResult,
+        });
+      }
+
+      console.log('Finished scanning files.');
+      console.log('\nDetailed results per file:');
+      console.dir(fileResults, { depth: null });
+
+      console.log('\nFiles per STIX Malware Analysis result:');
+      const resultValues = [
+        ...new Set([
+          ...STIX_MALWARE_RESULT_OV,
+          ...fileResults
+            .map((fileResult) => fileResult.stixMalwareAnalysis.result)
+            .sort(),
+        ]),
+      ];
+      resultValues.map((resultValue) => {
+        console.log('- ' + resultValue + ' :');
+        console.log(
+          fileResults
+            .filter(
+              (fileResult) =>
+                fileResult.stixMalwareAnalysis.result === resultValue,
+            )
+            .map((fileResult) => fileResult.file),
+        );
+      });
+
+      // TODO: Let storeMalwareAnalysis() run a SELECT query after the insert
+      //       to check in which (if any) graphs the resource was inserted?
+      //       Would be more reliable and easier to parse.
+      const filesNoDatabaseUpdate = fileResults.filter(
+        (fileResult) =>
+          !(
+            fileResult.databaseResponse?.results?.bindings[0]?.['callret-0']
+              ?.value &&
+            fileResult.databaseResponse?.results?.bindings[0]?.[
+              'callret-0'
+            ]?.value.match(' -- done')
+          ),
+      );
+      if (filesNoDatabaseUpdate.length) {
+        console.log(
+          '\nFiles for which the database response indicates that the ' +
+            'malware analysis resource object was not added to any graph:',
+        );
+        console.dir(filesNoDatabaseUpdate, { depth: null });
+      }
```

Comment:
I think this is really too much. At some point we should trust the system and the corresponding error codes if stuff fail.